### PR TITLE
Weights fixes (master)

### PR DESCRIPTION
**Pull request by MaelRL**
master version of https://github.com/CGAL/cgal/pull/6977.

5.5.x: https://github.com/CGAL/cgal/pull/7023.

ToDo:
- [x] Manually check uses of weights

### CODE REVIEW

**Comment by sloriot** [comment]
Errors in [CGAL-5.6-Ic-113](https://cgal.geometryfactory.com/CGAL/Members/testsuite/CGAL-5.6-Ic-113/Surface_mesh_deformation/TestReport_lrineau_Fedora.gz), related to OpenMesh

File: BGL/include/CGAL/boost/graph/properties_OpenMesh.h [hunk 1]
```diff
@@ -231,8 +233,8 @@ class OM_point_pmap
 #if defined(CGAL_USE_OM_POINTS)
     const_cast<OpenMesh&>(*pm.sm_).set_point(v,p);
 #else
-    const_cast<OpenMesh&>(*pm.sm_).set_point
-      (v, typename OpenMesh::Point((float)p[0], (float)p[1], (float)p[2]));
+    typedef typename CGAL::Kernel_traits<typename OpenMesh::Point>::type FT;
+    const_cast<OpenMesh&>(*pm.sm_).set_point(v, typename OpenMesh::Point(FT(p[0]), FT(p[1]), FT(p[2])));
```
Comment (MaelRL) [inline BGL/include/CGAL/boost/graph/properties_OpenMesh.h:237]: @sloriot @afabri A bit confused by the `float`: in the properties_OM, we use `EPICK::Point_3` when not using `OpenMesh::Point`. Was it to have symmetry with results obtained with OpenMesh for when only `float`-based `OpenMesh::Point`s were used?

Comment (sloriot) [reply BGL/include/CGAL/boost/graph/properties_OpenMesh.h:237]: Seems so:
https://www.graphics.rwth-aachen.de/media/openmesh_static/Documentations/OpenMesh-6.0-Documentation/a00131.html#a57c3a0075b2f50719679de132aa772b1

Comment (MaelRL) [reply BGL/include/CGAL/boost/graph/properties_OpenMesh.h:237]: It still is, but there also exists `DefaultTraitsDouble`, where `Point` is `vec3d`, which we do use in just a few places. In this case, we cast to float even though our Point is EPICK::Point_3 and the OpenMesh's scalar type is double.

**Comment by sloriot** [comment]
```
/mnt/testsuite/include/CGAL/boost/graph/properties_OpenMesh.h:234:84: error: no class template named ‘vector_traits’ in ‘class OpenMesh::PolyMesh_ArrayKernelT<>’
  234 |     typedef typename OpenMesh::vector_traits<typename OpenMesh::Point>::value_type Scalar;
```

**Comment by sloriot** [comment]
`decltype(pt.x())` ? or something like that?

**Comment by sloriot** [comment]
```
/home/cgal_tester/build/src/cmake/platforms/Fedora-with-LEDA/test/Surface_mesh_deformation/Cactus_deformation_session_OpenMesh.cpp:16:51: error: ‘DefaultTraitsDouble’ is not a member of ‘OpenMesh’; did you mean ‘DefaultTraits’?
   16 | typedef OpenMesh::PolyMesh_ArrayKernelT<OpenMesh::DefaultTraitsDouble> Mesh;
```

**Comment by sloriot** [comment]
I have on my machine:
```
drwxr-xr-x 9 sloriot sloriot 4096 Feb  2  2021 OpenMesh-OpenMesh-6.3
drwxr-xr-x 8 sloriot sloriot 4096 Feb  2  2021 OpenMesh-OpenMesh-7.1
drwxr-xr-x 8 sloriot sloriot 4096 Feb  2  2021 OpenMesh-OpenMesh-8.1
```

and I can find DefaultTraitsDouble only in 8.1

**Comment by afabri** [comment]
I could live with the requirement of 8.1

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-126

**Comment by lrineau** [comment]
@MaelRL @sloriot Should the `TODO` block the merge of this set of pull-requests?

**Comment by MaelRL** [comment]
Yes

**Comment by MaelRL** [comment]
I checked parameterization, skeletonization, heat method, hole filling, shape smoothing on a couple inputs and did not notice any regression visually.

**Pull request merged**